server/game/cards/06-TLS/MichaelTheBadgerDodge.js

```diff
@@ -0,0 +1,75 @@
+const DudeCard = require('../../dudecard.js');
+const GameActions = require('../../GameActions/index.js');
+
+/** @typedef {import('../../AbilityDsl')} AbilityDsl */
+
+class MichaelTheBadgerDodge extends DudeCard {
+    /** @param {AbilityDsl} ability */
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            targetController: 'current',
+            condition: () => this.hasAttachmentWithKeywords(['melee', 'weapon']), 
+            match: this,
+            effect: [
+                ability.effects.setAsStud()
+            ]
+        });
+        this.action({
+            title: 'Michael "The Badger" Dodge',
+            playType: ['shootout'],
+            cost: [
+                ability.costs.bootSelf(),
+                // Not strictly a cost, but needed so we can reference the result in the 
+                // action handler.
+                ability.costs.pull()
```

Comment:
I am not really against it being a cost, but if you do not want it, you can implement it similar to for example `TheRDRanch.js`

Reply:
I like the flow better when its not a cost (the pull message occurs in a more natural order in the chat log) so I switched it. Thanks for the pointer.